symplyphysics/laws/optics/interference_films_from_angle_refraction_refractive_index_wavelength.py

```diff
@@ -0,0 +1,57 @@
+from sympy import (Eq, solve, cos)
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+    dimensionless,
+    angle_type
+)
+
+# Description
+## Interference in thin films is a phenomenon that occurs as a result of the separation of a light beam when
+## reflected from the upper and lower boundaries of a thin film. As a result, there are two light waves that
+## can interfere.
+## The order of interference indicates the number of whole wavelengths. When the difference in the path of the
+## beam reflected from the upper boundary of the film and the beam reflected from the lower boundary of the
+## film is equal to an integer wave "n", then these two waves enter the observation point with the same phases
+## and demonstrate interference.
+## The angle of refraction of light is the angle between the ray passing through the surface and the normal to the surface.
+
+## Law is: h = k * L / (2 * n * sin(O)), where
```

Comment:
Could you show me, where you've got this formula?

Reply:
> Could you show me, where you've got this formula?

http://fn.bmstu.ru/data-physics/library/physbook/tom4/ch4/texthtml/ch4_4.htm

I have already noticed that there should be a cosine. I'll fix it.

Reply:
Do you mean formula 4.24b: 2nd * cos(theta2) = mL?
But it is a condition of minimum of interference. You've never mention that we expect minimum of interference in the law.

Reply:
> Do you mean formula 4.24b: 2nd * cos(theta2) = mL? But it is a condition of minimum of interference. You've never mention that we expect minimum of interference in the law.

If instead of m we substitute m = 1, 2, 3, ..., then this is the minimum condition. If instead of m we substitute m = 1/2, 3/2, 5/2, ..., then this is the maximum condition.

I can add this to the description. Is it okay?

Reply:
Not really. You've set the order of interference type to `int` so we cannot substitute it for non integer number. Therefore you describe the minimum condition in your law.
We can set order of interference to `float` though. This way we can describe both minumum and maximum conditions with this law. But what if someone puts some number that is not 1/2 or 1, ie 0.3?
We have two options here:
1. Throw an exception if order of interference is not integer and is not divisible by 2.
2. Use half wavelengths and always use integer number for order of interference. But it may confuse users who expected full wavelengths in this formula.